benchmarks/bench_softmax_ref4282.py

```
@@ -0,0 +1,193 @@
+        f"{'shape':<16} {'ref_ms':>10} {'ours_ms':>10} {'speedup':>8} {'ref_route':>10}"
+    )
+    speedups = []
+    for s in shapes:
+        rm = ref_rows[s]["median_ms"]
+        om = ours_rows[s]["median_ms"]
+        sp = rm / om
+        speedups.append(sp)
+        route = ROUTE_NAMES.get(ref_rows[s].get("route") or -1, "-")
+        marker = "" if ref_rows[s]["pass"] and ours_rows[s]["pass"] else "  FAIL"
+        print(
+            f"{str(list(s)):<16} {rm:>10.4f} {om:>10.4f} {sp:>7.3f}x {route:>10}{marker}"
```

Ruff (0.16.2)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)